alibi/explainers/partial_dependence.py

```diff
@@ -0,0 +1,1294 @@
+            features = (features, )
+
+        if grid_points is None:
+            grid_points = {}
+
+        deciles, values, features_indices = [], [], [],
+        for f in features:  # type: ignore[union-attr]
+            # Note that we are using the _safe_indexing procedure implement in sklearn to retrieve the column.
+            # This is because the _safe_indexing supports many data types such as sparse matrix representation,
+            # pandas dataframes etc. In case we would like to enhance alibi in the future to support other
+            # data types than np.ndarrays, this can be useful.
+            f_indices = np.asarray(_get_column_indices(X, f), dtype=np.int32, order='C').ravel()
+            X_f = _safe_indexing(X, f_indices, axis=1)
+
+            # get deciles for the current feature if the feature is numerical
+            deciles_f = get_quantiles(X_f, num_quantiles=11) if self._is_numerical(f) else None
+
+            if f not in grid_points:
+                # construct grid for feature f. Note that for categorical features we pass the
+                # grid resolution to be infinity because otherwise we risk to apply `linspace` to
+                # categorical values, which does not make sense.
+                _, values_f = _grid_from_X(X_f,
+                                           percentiles=percentiles,
+                                           grid_resolution=grid_resolution if self._is_numerical(f) else np.inf)
+            else:
+                values_f = [grid_points[f_indices[0]]]
+
+            features_indices.append(f_indices)
+            deciles.append(deciles_f)
+            values += values_f
+
+        # covers also the case of a single feature, just to ensure it has the right shape
+        features_indices = np.concatenate(features_indices, axis=0)
+        grid = cartesian(tuple([v.reshape(-1) for v in values]))
+
+        if method == "brute":
+            averaged_predictions, predictions = _partial_dependence_brute(
+                estimator, grid, features_indices, X, response_method
+            )
+
+            # reshape predictions to (n_outputs, n_instances, n_values_feature_0, n_values_feature_1, ...)
+            predictions = predictions.reshape(
+                -1, X.shape[0], *[val.shape[0] for val in values]
+            )
+        else:
+            averaged_predictions = _partial_dependence_recursion(
+                estimator, grid, features_indices
+            )
+
+        # reshape averaged_predictions to (n_outputs, n_values_feature_0, n_values_feature_1, ...)
+        averaged_predictions = averaged_predictions.reshape(
+            -1, *[val.shape[0] for val in values]
+        )
+
+        # special case when dealing with a binary classifier that uses predict_proba. sklearn removes
+        # the computation for the label 0 and only returns the predictions for label 1. To make it consistent with
+        # the ALE, we re-add the predictions for label 0.
```

Comment:
Would be good to add if there are concrete examples of such classifiers?

Reply:
Not sure I follow ... This happens for all classifiers that support `predict_proba` -- lots of classifiers. Can you elaborate?

Reply:
Ah I understand, so any 2-class classifier with `predict_proba` has this behaviour.